recipe/meta.yaml

```diff
@@ -181,6 +181,7 @@ outputs:
         - ${CC} -Wall tests/aligned_alloc.c -c -o c_aligned.o -v -fsanitize=address
         - ${CC} -Wall tests/aligned_alloc.c -c -o c_aligned.o -v
         - ${CC} -Wall c_aligned.o -o c_aligned -v && ./c_aligned   # [cross_target_platform == target_platform]
+        - ${CC} -Wall c_aligned.o -o c_aligned -Wl,-rpath,/foo && {{ gcc_machine }}-{{ gcc_vendor }}-linux-gnu-readelf -d c_aligned | grep RPATH | grep "/foo:${PREFIX}/lib"  # [cross_target_platform == target_platform]
```

Comment:
@isuruf Looking at this test, do you have the selector on it because you copy pasted and just left it there?  Seems like the test above is restricted to `cross == target` because it is executing a binary on target.  The rpath check should probably happen on all builds because we're tweaking the specs to do that on all configs, right?

Reply:
We probably shouldn't add the rpath change in the spec file for `cross != target`